ibm/resource_ibm_push_notification_chrome.go

```diff
@@ -0,0 +1,112 @@
+package ibm
+
+import (
+	"fmt"
+
+	"github.com/IBM/push-notifications-go-sdk/pushservicev1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceIBMPNApplicationChrome() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceApplicationChromeRead,
+		Create: resourceApplicationChromeCreate,
+		Update: resourceApplicationChromeUpdate,
+		Delete: resourceApplicationChromeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique ID of the application using the push service.",
+			},
+			"api_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "An API key that gives the push service an authorized access to Google services that is used for Chrome Web Push.",
+			},
+			"web_site_url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The URL of the WebSite / WebApp that should be permitted to subscribe to WebPush.",
+			},
+		},
+	}
+}
+
+func resourceApplicationChromeCreate(d *schema.ResourceData, meta interface{}) error {
+	pnClient, err := meta.(ClientSession).PushServiceV1()
+	if err != nil {
+		return err
+	}
+
+	serverKey := d.Get("api_key").(string)
+	websiteURL := d.Get("web_site_url").(string)
+	applicationGUID := d.Get("application_id").(string)
+
+	_, response, err := pnClient.SaveChromeWebConf(&pushservicev1.SaveChromeWebConfOptions{
+		ApplicationID: &applicationGUID,
+		ApiKey:        &serverKey,
+		WebSiteURL:    &websiteURL,
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error configuring chrome web platform: %s with responce code  %d", err, response.StatusCode)
+	}
+
+	return resourceApplicationChromeRead(d, meta)
+}
+
+func resourceApplicationChromeUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	if d.HasChanges("api_key", "web_site_url") {
+		return resourceApplicationChromeCreate(d, meta)
+	}
+	return nil
+}
+
+func resourceApplicationChromeRead(d *schema.ResourceData, meta interface{}) error {
+	pnClient, err := meta.(ClientSession).PushServiceV1()
+	if err != nil {
+		return err
+	}
+
+	applicationGUID := d.Get("application_id").(string)
```

Comment:
in create you need to set the ID...and in read you need to use d.ID() to read the values ..
Don't use d.Get() in read methods

Reply:
done